check-cf-space-quota-utilization/README.md

```diff
@@ -3,17 +3,178 @@
+
+* Executor ```getCfOrg``` runs a ```GET``` request to the ```/v3/organizations/:guid``` endpoint to acquire the Cloud Foundry Organization Quota GUID.
+
+* Executor ```getCfOrgUsage``` runs a ```GET``` request to the ```/v3/organizations/:guid/usage_summary``` to acquire the specified Cloud Foundry Organization object's memory and app instance usage summary.
+
+* Executor ```getCfOrgQuota``` runs a ```GET``` request to the ```/v3/organization_quotas/:guid``` endpoint to acquire the quota details of the Cloud Foundry Organization.
+
+* Executor ```getServiceInstanceByOrg``` runs a ```GET``` request to the ```/v3/service_instances/:guid``` endpoint to acquire all the instances contained in the specific Cloud Foundry Organization.
+
+* Executor ```getCfSpaceQuota``` runs a ```GET``` request to the ```/v3/space_quotas/:guid``` endpoint to acquire the Cloud Foundry Space Quota.
+
+* Executor ```getServiceInstanceBySpace``` runs a ```GET``` request to the ```/v3/service_instances?space_guids=<space-guid>``` request with the ```space_guids``` parameter to fetch all the service instances in a space.
+
+* Executor ```getCfApps``` relies on the ```ListCfApps``` command that lists all the Cloud Foundry application in a selected space. 
+
+* Executor ```calculateSpaceMemory```  relies on the input key ```appResources``` that consists of the resourceNames of all the applications in a given space. This input key is based on the information received when executing the ```ListCfApps``` command. ```calculateSpaceMemory``` then calculates the memory usage of every single application by using the ```calculateAppMemory```.
+
+* Executor ```calculateSpaceMemoryPercentage``` calculates the space memory usage in percentages.
+
+* Executor ```calculateOrgMemoryPercentage``` calculates the organization memory usage in percentages.
+
+* Executor ```SendAnsEvent``` send the event data to SAP Alert Notification Service where it would be processed according to your configuration.
+
+### Event Details
+
+| Event Property | Value |
+| -------------- | ----- |
+| eventType | **ResourceQuotaUtilization** |
+| severity | **INFO** |
+| category | **NOTIFICATION** |
+| priority | n/a |
+| subject | The subject is a constant value: **Space Quota Utilization** |
+| body | The value of this parameter is the following text:<br>**Quota Utilization for space <space name>:**<br>**Memory Total (MB):** <used memory in space> of <memory quota for the space> [<percentage>%]<br>**Service Count:** <number of services in space> of <service quota for the space> [<percentage>%]<br>**Quota Utilization for organization <organization name>:**<br>**Memory Total (MB):** <total used memory in organization> of <memory quota for the organization> [<percentage>%]<br>**Service Count:** <total number of services in organization> of <service quota for the organization> [<percentage>%]<br>*Calculation is based on utilization of all spaces for which authorization is explicitly granted.* |
+| resource | ... |
```

Comment:
We can additionally describe the resource tags here as well (e.g. space name and org name)